swift/ql/lib/codeql/swift/elements/expr/ApplyExpr.qll

```diff
@@ -9,6 +9,9 @@ class ApplyExpr extends ApplyExprBase {
     result = "call to " + this.getStaticTarget().toString()
     or
     not exists(this.getStaticTarget()) and
+    result = this.getFunction().toString()
```

Comment:
I like the output a lot more now, but I'm not sure we're getting the improved output for the right reasons. It looks like we're now generating `"call to f()"` when we can resolve the call to a static call target `f`, but we print just `"f()"` when we cannot resolve it to a target.

The reason the output looks a lot better now is that the cases where there isn't a static call target are all examples of the form where there's a strange `DotSyntaxCallExpr` (or `SelfApplyExpr`, or a `ConstructorRefCallExpr`) which all have strings that look like `"call to f()"` (because they fall into the category of calls where we can resolve the target).

For example, I think the call to `bar()` in the example below will get a confusing `toString` (since it'll just be the string `"bar"` and not `"call to bar"` even though it's a call to bar and not a reference to the variable).

```swift
func foo(_ bar: () -> ()) {
  bar()
}

foo({ print("I am a lambda :wave:") })
```

Does that make sense?

Reply:
Thanks for this, I need to think about it a bit more.  But since I've factored out this code for a future PR now, it is no longer an issue holding us up here.